oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -160,8 +265,18 @@ public Server getNode(String id) {
 
    @Override
    public void destroyNode(String id) {
-      waitServerUntilAvailable.apply(getNode(id));
+      Server node = getNode(id);
+      waitServerUntilAvailable.apply(node);
       logger.debug("Destroying %s ...", id);
+      try {
+         GenericQueryOptions options = new GenericQueryOptions().options(0, 0, null, node.name() + " firewall policy", null);
+         List<FirewallPolicy> firewallRules = api.firewallPolicyApi().list(options);
+         for (FirewallPolicy firewallRule : firewallRules) {
+            api.firewallPolicyApi().delete(firewallRule.id());
+         }
+      } catch (Exception ex) {
+         logger.debug("no firewall policies found for %s ...", id);
+      }
```

Comment:
The logic to delete the firewalls should be better moved to a `OneAndOneComputeService` class that implements the `cleanUpIncidentalResourcesOfDeadNodes` method. Thake the [azurecompute-arm implementation](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeService.java#L103) as an example.